### PR TITLE
Use logging in new_release

### PR DESCRIPTION
**Pull request by mwoehlke-kitware**
Actually use logging in new_release script rather than using bare print statements everywhere. Also, use f-strings in more (most) places.

- - -
This change is [Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20740)

### CODE REVIEW

**Comment by mwoehlke-kitware** [comment]
@jwnimmer-tri, do you want to take a look at this, or should I assign someone else? (I assume single review is probably also okay?)

...I was annoyed that 90-95% of the output (~130 lines of noise out of ~140 total) was skipping rust stuff or  ignoring pre-release tags.

**Review by jwnimmer-tri — COMMENTED** [review]
In general, I'm 100% in support of using `logging` over `print`.  I posted one feature-review comment below that needs a look.

After that's fixed, feel free to assign the on-call platform reviewer to finish it out.

> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20740#-:-NnLP0zP0tOxfEE9uPkU:bmhs1mm)* status: 1 unresolved discussion, needs platform reviewer assigned, needs at least two assigned reviewers

___
*[`tools/workspace/new_release.py` line 136 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/20740#-NnLQSRgBFKtViNVAcmX:-NnLQSRgBFKtViNVAcmY:b-iyuqj2) ([raw file](https://github.com/RobotLocomotion/drake/blob/5c1a84fab354be117dde7a7539a17cfd8047934b/tools/workspace/new_release.py#L136)):*
> ```Python
>     prerelease = any(stage in commit for stage in development_stages)
>     if prerelease:
>         logging.debug(f"Skipping prerelease {commit} for {workspace}")
> ```

It's very important that new_release _always_ finds a new release when there is one, since we _never_ check for new releases by hand as a cross-check.  The false negative rate must be 0%.

Therefore, anytime we "skip" a release we must either be absolutely 100% sure that it was _supposed_ to be ignored, or else we need to tell the user about the name and let _them_ be the one who OK's the skipping over it.  That's why we had these printouts, so that the user would notice any suspicios skips.

In other words, "debug" is too quiet for this situation.  If there is too much spam from this line, then we should instead add _much more specific_ exclusions (e.g., exclusions specific to particular repositories one-by-one).

**Review by mwoehlke-kitware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20740)* status: 1 unresolved discussion, needs platform reviewer assigned, needs at least two assigned reviewers

___
*[`tools/workspace/new_release.py` line 136 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/20740#-NnLQSRgBFKtViNVAcmX:-NnLnXWf6n7rz1R7nV9G:blvt1zg) ([raw file](https://github.com/RobotLocomotion/drake/blob/5c1a84fab354be117dde7a7539a17cfd8047934b/tools/workspace/new_release.py#L136)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

It's very important that new_release _always_ finds a new release when there is one, since we _never_ check for new releases by hand as a cross-check.  The false negative rate must be 0%.

Therefore, anytime we "skip" a release we must either be absolutely 100% sure that it was _supposed_ to be ignored, or else we need to tell the user about the name and let _them_ be the one who OK's the skipping over it.  That's why we had these printouts, so that the user would notice any suspicios skips.

In other words, "debug" is too quiet for this situation.  If there is too much spam from this line, then we should instead add _much more specific_ exclusions (e.g., exclusions specific to particular repositories one-by-one).
</blockquote></details>

Well, right now this is *extremely* noisy; indeed, it's what motivated me to make the change in the first place...

```
$ bazel run //tools/workspace:new_release  
INFO: Analyzed target //tools/workspace:new_release (0 packages loaded, 0 targets configured).  
INFO: Found 1 target...  
Target //tools/workspace:new_release up-to-date:  
 bazel-bin/tools/workspace/new_release  
INFO: Elapsed time: 0.259s, Critical Path: 0.00s  
INFO: 1 process: 1 internal.  
INFO: Build completed successfully, 1 total action  
INFO: Running command line: bazel-bin/tools/workspace/new_release  
Collecting bazel repository details...  
Loading: 0 packages loaded  
Loading: 340 packages loaded  
INFO: All external dependencies fetched successfully.  
Loading: 0 packages loaded  
Checking for new releases...  
abseil_cpp_internal needs upgrade from 8028a87c96df0fff5ab58daeec30c43ce6fb0d20 to d5a2cec006d14c6801ddeb768bf2574a1cf4fa7f  
Ignoring crate__amd-0.2.2 from rules_rust  
Ignoring crate__autocfg-1.1.0 from rules_rust  
Ignoring crate__blas-0.22.0 from rules_rust  
Ignoring crate__blas-sys-0.7.1 from rules_rust  
Ignoring crate__cfg-if-1.0.0 from rules_rust  
Ignoring crate__clarabel-0.6.0 from rules_rust  
Ignoring crate__darling-0.14.4 from rules_rust  
Ignoring crate__darling_core-0.14.4 from rules_rust  
Ignoring crate__darling_macro-0.14.4 from rules_rust  
Ignoring crate__derive_builder-0.11.2 from rules_rust  
Ignoring crate__derive_builder_core-0.11.2 from rules_rust  
Ignoring crate__derive_builder_macro-0.11.2 from rules_rust  
Ignoring crate__either-1.9.0 from rules_rust  
Ignoring crate__enum_dispatch-0.3.12 from rules_rust  
Ignoring crate__fnv-1.0.7 from rules_rust  
Ignoring crate__ident_case-1.0.1 from rules_rust  
Ignoring crate__itertools-0.11.0 from rules_rust  
Ignoring crate__lapack-0.19.0 from rules_rust  
Ignoring crate__lapack-sys-0.14.0 from rules_rust  
Ignoring crate__lazy_static-1.4.0 from rules_rust  
Ignoring crate__libc-0.2.149 from rules_rust  
Ignoring crate__num-complex-0.4.4 from rules_rust  
Ignoring crate__num-traits-0.2.17 from rules_rust  
Ignoring crate__once_cell-1.18.0 from rules_rust  
Ignoring crate__proc-macro2-1.0.69 from rules_rust  
Ignoring crate__quote-1.0.33 from rules_rust  
Ignoring crate__strsim-0.10.0 from rules_rust  
Ignoring crate__syn-1.0.109 from rules_rust  
Ignoring crate__syn-2.0.38 from rules_rust  
Ignoring crate__thiserror-1.0.50 from rules_rust  
Ignoring crate__thiserror-impl-1.0.50 from rules_rust  
Ignoring crate__unicode-ident-1.0.12 from rules_rust  
curl_internal needs upgrade from curl-8_4_0 to curl-8_5_0  
dm_control_internal needs upgrade from 1.0.15 to 1.0.16  
Skipping prerelease v1.7.0beta89 for libpng_internal  
Skipping prerelease v1.7.0beta88 for libpng_internal  
Skipping prerelease v1.7.0beta87 for libpng_internal  
Skipping prerelease v1.7.0beta86 for libpng_internal  
Skipping prerelease v1.7.0beta85 for libpng_internal  
Skipping prerelease v1.7.0beta84 for libpng_internal  
Skipping prerelease v1.7.0beta83 for libpng_internal  
Skipping prerelease v1.7.0beta82 for libpng_internal  
Skipping prerelease v1.7.0beta81 for libpng_internal  
Skipping prerelease v1.7.0beta80 for libpng_internal  
Skipping prerelease v1.7.0beta79 for libpng_internal  
Skipping prerelease v1.7.0beta78 for libpng_internal  
Skipping prerelease v1.7.0beta77 for libpng_internal  
Skipping prerelease v1.7.0beta76 for libpng_internal  
Skipping prerelease v1.7.0beta75 for libpng_internal  
Skipping prerelease v1.7.0beta74 for libpng_internal  
Skipping prerelease v1.7.0beta73 for libpng_internal  
Skipping prerelease v1.7.0beta72 for libpng_internal  
Skipping prerelease v1.7.0beta71 for libpng_internal  
Skipping prerelease v1.7.0beta70 for libpng_internal  
Skipping prerelease v1.7.0beta69 for libpng_internal  
Skipping prerelease v1.7.0beta68 for libpng_internal  
Skipping prerelease v1.7.0beta67 for libpng_internal  
Skipping prerelease v1.7.0beta66 for libpng_internal  
Skipping prerelease v1.7.0beta65 for libpng_internal  
Skipping prerelease v1.7.0beta64 for libpng_internal  
Skipping prerelease v1.7.0beta63 for libpng_internal  
Skipping prerelease v1.7.0beta62 for libpng_internal  
Skipping prerelease v1.7.0beta61 for libpng_internal  
Skipping prerelease v1.7.0beta60 for libpng_internal  
Skipping prerelease v1.7.0beta59 for libpng_internal  
Skipping prerelease v1.7.0beta58 for libpng_internal  
Skipping prerelease v1.7.0beta57 for libpng_internal  
Skipping prerelease v1.7.0beta56 for libpng_internal  
Skipping prerelease v1.7.0beta55 for libpng_internal  
Skipping prerelease v1.7.0beta54 for libpng_internal  
Skipping prerelease v1.7.0beta53 for libpng_internal  
Skipping prerelease v1.7.0beta52 for libpng_internal  
Skipping prerelease v1.7.0beta51 for libpng_internal  
Skipping prerelease v1.7.0beta50 for libpng_internal  
Skipping prerelease v1.7.0beta49 for libpng_internal  
Skipping prerelease v1.7.0beta48 for libpng_internal  
Skipping prerelease v1.7.0beta47 for libpng_internal  
Skipping prerelease v1.7.0beta46 for libpng_internal  
Skipping prerelease v1.7.0beta45 for libpng_internal  
Skipping prerelease v1.7.0beta44 for libpng_internal  
Skipping prerelease v1.7.0beta43 for libpng_internal  
Skipping prerelease v1.7.0beta42 for libpng_internal  
Skipping prerelease v1.7.0beta41 for libpng_internal  
Skipping prerelease v1.7.0beta40 for libpng_internal  
Skipping prerelease v1.7.0beta39 for libpng_internal  
Skipping prerelease v1.7.0beta38 for libpng_internal  
Skipping prerelease v1.7.0beta37 for libpng_internal  
Skipping prerelease v1.7.0beta36 for libpng_internal  
Skipping prerelease v1.7.0beta35 for libpng_internal  
Skipping prerelease v1.7.0beta34 for libpng_internal  
Skipping prerelease v1.7.0beta33 for libpng_internal  
Skipping prerelease v1.7.0beta32 for libpng_internal  
Skipping prerelease v1.7.0beta31 for libpng_internal  
Skipping prerelease v1.7.0beta30 for libpng_internal  
Skipping prerelease v1.7.0beta29 for libpng_internal  
Skipping prerelease v1.7.0beta28 for libpng_internal  
Skipping prerelease v1.7.0beta27 for libpng_internal  
Skipping prerelease v1.7.0beta26 for libpng_internal  
Skipping prerelease v1.7.0beta25 for libpng_internal  
Skipping prerelease v1.7.0beta24 for libpng_internal  
Skipping prerelease v1.7.0beta23 for libpng_internal  
Skipping prerelease v1.7.0beta22 for libpng_internal  
Skipping prerelease v1.7.0beta21 for libpng_internal  
Skipping prerelease v1.7.0beta20 for libpng_internal  
Skipping prerelease v1.7.0beta19 for libpng_internal  
Skipping prerelease v1.7.0beta18 for libpng_internal  
Skipping prerelease v1.7.0beta17 for libpng_internal  
Skipping prerelease v1.7.0beta16 for libpng_internal  
Skipping prerelease v1.7.0beta15 for libpng_internal  
Skipping prerelease v1.7.0beta14 for libpng_internal  
Skipping prerelease v1.7.0beta13 for libpng_internal  
Skipping prerelease v1.7.0beta12 for libpng_internal  
Skipping prerelease v1.7.0beta11 for libpng_internal  
Skipping prerelease v1.7.0beta10 for libpng_internal  
Skipping prerelease v1.7.0beta09 for libpng_internal  
Skipping prerelease v1.7.0beta08 for libpng_internal  
Skipping prerelease v1.7.0beta07 for libpng_internal  
Skipping prerelease v1.7.0beta06 for libpng_internal  
Skipping prerelease v1.7.0beta05 for libpng_internal  
Skipping prerelease v1.7.0beta04 for libpng_internal  
Skipping prerelease v1.7.0beta03 for libpng_internal  
Skipping prerelease v1.7.0beta02 for libpng_internal  
Skipping prerelease v1.7.0beta01 for libpng_internal  
Skipping prerelease v1.7.0alpha10 for libpng_internal  
Skipping prerelease v1.7.0alpha09 for libpng_internal  
Skipping prerelease v1.7.0alpha08 for libpng_internal  
Skipping prerelease v1.7.0alpha07 for libpng_internal  
Skipping prerelease v1.7.0alpha06 for libpng_internal  
Skipping prerelease v1.7.0alpha05 for libpng_internal  
Skipping prerelease v1.7.0alpha04 for libpng_internal  
Skipping prerelease v1.7.0alpha03 for libpng_internal  
Skipping prerelease v1.7.0alpha02 for libpng_internal  
Skipping prerelease v1.7.0alpha01 for libpng_internal  
mypy_internal needs upgrade from v1.7.1 to v1.8.0  
nanoflann_internal needs upgrade from v1.5.2 to v1.5.3  
rules_rust needs upgrade from 0.32.0 to 0.36.2  
Skipping prerelease sdformat-prerelease_2.0.3 for sdformat_internal  
Skipping prerelease sdformat-prerelease_2.0.2 for sdformat_internal  
Skipping prerelease sdformat-prerelease_2.0.1 for sdformat_internal  
Skipping prerelease sdformat-prerelease_2.0.0 for sdformat_internal  
Skipping prerelease sdformat-prerelease_1.5.0 for sdformat_internal  
Skipping prerelease sdformat-prerelease_1.4.9 for sdformat_internal  
suitesparse_internal needs upgrade from v7.3.1 to v7.4.0  
tinyobjloader_internal needs upgrade from ee45fb41db95bf9563f2a41bc63adfa18475c2ee to cc327eecf7f8f4139932aec8d75db2d091f412ef  
tinyxml2_internal needs upgrade from 9.0.0 to 10.0.0  
typing_extensions_internal needs upgrade from 4.8.0 to 4.9.0
```

**Review by jwnimmer-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20740)* status: 1 unresolved discussion, needs platform reviewer assigned, needs at least two assigned reviewers

___
*[`tools/workspace/new_release.py` line 136 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/20740#-NnLQSRgBFKtViNVAcmX:-NnLoiVZA2zCe8llYUTK:b-58gfin) ([raw file](https://github.com/RobotLocomotion/drake/blob/5c1a84fab354be117dde7a7539a17cfd8047934b/tools/workspace/new_release.py#L136)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

Well, right now this is *extremely* noisy; indeed, it's what motivated me to make the change in the first place...

```
$ bazel run //tools/workspace:new_release  
INFO: Analyzed target //tools/workspace:new_release (0 packages loaded, 0 targets configured).  
INFO: Found 1 target...  
Target //tools/workspace:new_release up-to-date:  
 bazel-bin/tools/workspace/new_release  
INFO: Elapsed time: 0.259s, Critical Path: 0.00s  
INFO: 1 process: 1 internal.  
INFO: Build completed successfully, 1 total action  
INFO: Running command line: bazel-bin/tools/workspace/new_release  
Collecting bazel repository details...  
Loading: 0 packages loaded  
Loading: 340 packages loaded  
INFO: All external dependencies fetched successfully.  
Loading: 0 packages loaded  
Checking for new releases...  
abseil_cpp_internal needs upgrade from 8028a87c96df0fff5ab58daeec30c43ce6fb0d20 to d5a2cec006d14c6801ddeb768bf2574a1cf4fa7f  
Ignoring crate__amd-0.2.2 from rules_rust  
Ignoring crate__autocfg-1.1.0 from rules_rust  
Ignoring crate__blas-0.22.0 from rules_rust  
Ignoring crate__blas-sys-0.7.1 from rules_rust  
Ignoring crate__cfg-if-1.0.0 from rules_rust  
Ignoring crate__clarabel-0.6.0 from rules_rust  
Ignoring crate__darling-0.14.4 from rules_rust  
Ignoring crate__darling_core-0.14.4 from rules_rust  
Ignoring crate__darling_macro-0.14.4 from rules_rust  
Ignoring crate__derive_builder-0.11.2 from rules_rust  
Ignoring crate__derive_builder_core-0.11.2 from rules_rust  
Ignoring crate__derive_builder_macro-0.11.2 from rules_rust  
Ignoring crate__either-1.9.0 from rules_rust  
Ignoring crate__enum_dispatch-0.3.12 from rules_rust  
Ignoring crate__fnv-1.0.7 from rules_rust  
Ignoring crate__ident_case-1.0.1 from rules_rust  
Ignoring crate__itertools-0.11.0 from rules_rust  
Ignoring crate__lapack-0.19.0 from rules_rust  
Ignoring crate__lapack-sys-0.14.0 from rules_rust  
Ignoring crate__lazy_static-1.4.0 from rules_rust  
Ignoring crate__libc-0.2.149 from rules_rust  
Ignoring crate__num-complex-0.4.4 from rules_rust  
Ignoring crate__num-traits-0.2.17 from rules_rust  
Ignoring crate__once_cell-1.18.0 from rules_rust  
Ignoring crate__proc-macro2-1.0.69 from rules_rust  
Ignoring crate__quote-1.0.33 from rules_rust  
Ignoring crate__strsim-0.10.0 from rules_rust  
Ignoring crate__syn-1.0.109 from rules_rust  
Ignoring crate__syn-2.0.38 from rules_rust  
Ignoring crate__thiserror-1.0.50 from rules_rust  
Ignoring crate__thiserror-impl-1.0.50 from rules_rust  
Ignoring crate__unicode-ident-1.0.12 from rules_rust  
curl_internal needs upgrade from curl-8_4_0 to curl-8_5_0  
dm_control_internal needs upgrade from 1.0.15 to 1.0.16  
Skipping prerelease v1.7.0beta89 for libpng_internal  
Skipping prerelease v1.7.0beta88 for libpng_internal  
Skipping prerelease v1.7.0beta87 for libpng_internal  
Skipping prerelease v1.7.0beta86 for libpng_internal  
Skipping prerelease v1.7.0beta85 for libpng_internal  
Skipping prerelease v1.7.0beta84 for libpng_internal  
Skipping prerelease v1.7.0beta83 for libpng_internal  
Skipping prerelease v1.7.0beta82 for libpng_internal  
Skipping prerelease v1.7.0beta81 for libpng_internal  
Skipping prerelease v1.7.0beta80 for libpng_internal  
Skipping prerelease v1.7.0beta79 for libpng_internal  
Skipping prerelease v1.7.0beta78 for libpng_internal  
Skipping prerelease v1.7.0beta77 for libpng_internal  
Skipping prerelease v1.7.0beta76 for libpng_internal  
Skipping prerelease v1.7.0beta75 for libpng_internal  
Skipping prerelease v1.7.0beta74 for libpng_internal  
Skipping prerelease v1.7.0beta73 for libpng_internal  
Skipping prerelease v1.7.0beta72 for libpng_internal  
Skipping prerelease v1.7.0beta71 for libpng_internal  
Skipping prerelease v1.7.0beta70 for libpng_internal  
Skipping prerelease v1.7.0beta69 for libpng_internal  
Skipping prerelease v1.7.0beta68 for libpng_internal  
Skipping prerelease v1.7.0beta67 for libpng_internal  
Skipping prerelease v1.7.0beta66 for libpng_internal  
Skipping prerelease v1.7.0beta65 for libpng_internal  
Skipping prerelease v1.7.0beta64 for libpng_internal  
Skipping prerelease v1.7.0beta63 for libpng_internal  
Skipping prerelease v1.7.0beta62 for libpng_internal  
Skipping prerelease v1.7.0beta61 for libpng_internal  
Skipping prerelease v1.7.0beta60 for libpng_internal  
Skipping prerelease v1.7.0beta59 for libpng_internal  
Skipping prerelease v1.7.0beta58 for libpng_internal  
Skipping prerelease v1.7.0beta57 for libpng_internal  
Skipping prerelease v1.7.0beta56 for libpng_internal  
Skipping prerelease v1.7.0beta55 for libpng_internal  
Skipping prerelease v1.7.0beta54 for libpng_internal  
Skipping prerelease v1.7.0beta53 for libpng_internal  
Skipping prerelease v1.7.0beta52 for libpng_internal  
Skipping prerelease v1.7.0beta51 for libpng_internal  
Skipping prerelease v1.7.0beta50 for libpng_internal  
Skipping prerelease v1.7.0beta49 for libpng_internal  
Skipping prerelease v1.7.0beta48 for libpng_internal  
Skipping prerelease v1.7.0beta47 for libpng_internal  
Skipping prerelease v1.7.0beta46 for libpng_internal  
Skipping prerelease v1.7.0beta45 for libpng_internal  
Skipping prerelease v1.7.0beta44 for libpng_internal  
Skipping prerelease v1.7.0beta43 for libpng_internal  
Skipping prerelease v1.7.0beta42 for libpng_internal  
Skipping prerelease v1.7.0beta41 for libpng_internal  
Skipping prerelease v1.7.0beta40 for libpng_internal  
Skipping prerelease v1.7.0beta39 for libpng_internal  
Skipping prerelease v1.7.0beta38 for libpng_internal  
Skipping prerelease v1.7.0beta37 for libpng_internal  
Skipping prerelease v1.7.0beta36 for libpng_internal  
Skipping prerelease v1.7.0beta35 for libpng_internal  
Skipping prerelease v1.7.0beta34 for libpng_internal  
Skipping prerelease v1.7.0beta33 for libpng_internal  
Skipping prerelease v1.7.0beta32 for libpng_internal  
Skipping prerelease v1.7.0beta31 for libpng_internal  
Skipping prerelease v1.7.0beta30 for libpng_internal  
Skipping prerelease v1.7.0beta29 for libpng_internal  
Skipping prerelease v1.7.0beta28 for libpng_internal  
Skipping prerelease v1.7.0beta27 for libpng_internal  
Skipping prerelease v1.7.0beta26 for libpng_internal  
Skipping prerelease v1.7.0beta25 for libpng_internal  
Skipping prerelease v1.7.0beta24 for libpng_internal  
Skipping prerelease v1.7.0beta23 for libpng_internal  
Skipping prerelease v1.7.0beta22 for libpng_internal  
Skipping prerelease v1.7.0beta21 for libpng_internal  
Skipping prerelease v1.7.0beta20 for libpng_internal  
Skipping prerelease v1.7.0beta19 for libpng_internal  
Skipping prerelease v1.7.0beta18 for libpng_internal  
Skipping prerelease v1.7.0beta17 for libpng_internal  
Skipping prerelease v1.7.0beta16 for libpng_internal  
Skipping prerelease v1.7.0beta15 for libpng_internal  
Skipping prerelease v1.7.0beta14 for libpng_internal  
Skipping prerelease v1.7.0beta13 for libpng_internal  
Skipping prerelease v1.7.0beta12 for libpng_internal  
Skipping prerelease v1.7.0beta11 for libpng_internal  
Skipping prerelease v1.7.0beta10 for libpng_internal  
Skipping prerelease v1.7.0beta09 for libpng_internal  
Skipping prerelease v1.7.0beta08 for libpng_internal  
Skipping prerelease v1.7.0beta07 for libpng_internal  
Skipping prerelease v1.7.0beta06 for libpng_internal  
Skipping prerelease v1.7.0beta05 for libpng_internal  
Skipping prerelease v1.7.0beta04 for libpng_internal  
Skipping prerelease v1.7.0beta03 for libpng_internal  
Skipping prerelease v1.7.0beta02 for libpng_internal  
Skipping prerelease v1.7.0beta01 for libpng_internal  
Skipping prerelease v1.7.0alpha10 for libpng_internal  
Skipping prerelease v1.7.0alpha09 for libpng_internal  
Skipping prerelease v1.7.0alpha08 for libpng_internal  
Skipping prerelease v1.7.0alpha07 for libpng_internal  
Skipping prerelease v1.7.0alpha06 for libpng_internal  
Skipping prerelease v1.7.0alpha05 for libpng_internal  
Skipping prerelease v1.7.0alpha04 for libpng_internal  
Skipping prerelease v1.7.0alpha03 for libpng_internal  
Skipping prerelease v1.7.0alpha02 for libpng_internal  
Skipping prerelease v1.7.0alpha01 for libpng_internal  
mypy_internal needs upgrade from v1.7.1 to v1.8.0  
nanoflann_internal needs upgrade from v1.5.2 to v1.5.3  
rules_rust needs upgrade from 0.32.0 to 0.36.2  
Skipping prerelease sdformat-prerelease_2.0.3 for sdformat_internal  
Skipping prerelease sdformat-prerelease_2.0.2 for sdformat_internal  
Skipping prerelease sdformat-prerelease_2.0.1 for sdformat_internal  
Skipping prerelease sdformat-prerelease_2.0.0 for sdformat_internal  
Skipping prerelease sdformat-prerelease_1.5.0 for sdformat_internal  
Skipping prerelease sdformat-prerelease_1.4.9 for sdformat_internal  
suitesparse_internal needs upgrade from v7.3.1 to v7.4.0  
tinyobjloader_internal needs upgrade from ee45fb41db95bf9563f2a41bc63adfa18475c2ee to cc327eecf7f8f4139932aec8d75db2d091f412ef  
tinyxml2_internal needs upgrade from 9.0.0 to 10.0.0  
typing_extensions_internal needs upgrade from 4.8.0 to 4.9.0
```
</blockquote></details>

I'm not disputing that it's noisy and would be an improvement to reduce the noise.  I'm only asking that whatever solution gets applied doesn't have false negatives.

Another idea would be to compare date of the putative prerelease to the current stable release's date, or the selected new-stable releases's date, and then mute pre-releases that have already been superseded.

**Review by jwnimmer-tri — COMMENTED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20740)* status: 2 unresolved discussions, LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers

___
*[`tools/workspace/new_release.py` line 447 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/20740#-Npe7QXU5AsKC-5-eImu:-Npe7QXVAtk-4zmv2cuX:b-fslhcg) ([raw file](https://github.com/RobotLocomotion/drake/blob/5c1a84fab354be117dde7a7539a17cfd8047934b/tools/workspace/new_release.py#L447)):*
> ```Python
> 
>     # Log in to github.
>     if args.use_password:
> ```

BTW Does anyone use `--use_password` option that you know of?  I've been thinking we should rip it out (to simplify the tool's command-line arguments and overview docs).
___
*[`tools/workspace/new_release.py` line 482 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/20740#-Npe8Noo4ZziqJsk6yT5:-Npe8Noo4ZziqJsk6yT6:b-lbhx5o) ([raw file](https://github.com/RobotLocomotion/drake/blob/5c1a84fab354be117dde7a7539a17cfd8047934b/tools/workspace/new_release.py#L482)):*
> ```Python
>     metadata = read_repository_metadata(repositories=workspaces)
>     if args.verbose:
>         logging.debug(json.dumps(metadata, sort_keys=True, indent=2))
> ```

nit The logger object already handles verbosity control.  (The only thing `args.verbose` should be doing anymore is changing how the logger object gets configured.)



_Suggestion:_
```Python
     logging.getLogger('new_release').debug(json.dumps(metadata, sort_keys=True, indent=2))
```

**Review by mwoehlke-kitware — COMMENTED** [review]
+@jwnimmer-tri for feature review, please.

> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20740#-:-Nq-WVk6EGkf6U0AsiDS:b-qf68jg)* status: 2 unresolved discussions, LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers

___
*[`tools/workspace/new_release.py` line 447 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/20740#-Npe7QXU5AsKC-5-eImu:-Nq-W-6X4NOtQM_WLyhv:bx7k5xk) ([raw file](https://github.com/RobotLocomotion/drake/blob/5c1a84fab354be117dde7a7539a17cfd8047934b/tools/workspace/new_release.py#L447)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

BTW Does anyone use `--use_password` option that you know of?  I've been thinking we should rip it out (to simplify the tool's command-line arguments and overview docs).
</blockquote></details>

Probably fine; let's do it in a separate PR, though. :slightly_smiling_face: 
___
*[`tools/workspace/new_release.py` line 482 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/20740#-Npe8Noo4ZziqJsk6yT5:-Nq-W8ws68KX3qcxUmut:bdoe3pr) ([raw file](https://github.com/RobotLocomotion/drake/blob/5c1a84fab354be117dde7a7539a17cfd8047934b/tools/workspace/new_release.py#L482)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

nit The logger object already handles verbosity control.  (The only thing `args.verbose` should be doing anymore is changing how the logger object gets configured.)
</blockquote></details>

I'd as soon not format a rather large chunk of JSON if we aren't going to print it. I've implemented one solution that avoids the `if`, as suggested, however I think it's probably better to just write:
```python
    if logging.isEnabledFor(logging.DEBUG):
        logging.debug(json.dumps(metadata, sort_keys=True, indent=2))
```
WDYT?

**Review by jwnimmer-tri — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20740)* status: 1 unresolved discussion, LGTM missing from assignee jwnimmer-tri(platform), needs at least two assigned reviewers

___
*[`tools/workspace/new_release.py` line 482 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/20740#-Npe8Noo4ZziqJsk6yT5:-Nq-it708yEYTfzgCwCl:b-uc3pmf) ([raw file](https://github.com/RobotLocomotion/drake/blob/5c1a84fab354be117dde7a7539a17cfd8047934b/tools/workspace/new_release.py#L482)):*
> I'd as soon not format a rather large chunk of JSON if we aren't going to print it..

But **why**?  Stringizing the JSON takes on the order of 1 millsecond.  Computing the JSON we're going to dump takes on the order of seconds.  Any performance considerations here seem like a giant nothingburger.  We should keep the code as simple as practical, and I don't see any reason why wasting 1ms matters at all for this too.

**Review by jwnimmer-tri — COMMENTED** [review]
:lgtm: feature.

+@xuchenhan-tri for platform review per schedule, please.

> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20740#-:-Nq3qC48ENYcgNdALuSW:brw9p00)* status: LGTM missing from assignee xuchenhan-tri(platform)

**Review by xuchenhan-tri — COMMENTED** [review]
:lgtm: 

> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20740#-:-Nq3vTZI1LhJNes6ePtS:bg7igkv)* status: <span>:shipit:</span> complete! all discussions resolved, LGTM from assignees jwnimmer-tri(platform),xuchenhan-tri(platform)
